include/nbl/builtin/glsl/blit/blit/blit.glsl

```diff
@@ -0,0 +1,177 @@
+#ifndef _NBL_GLSL_BLIT_INCLUDED_
+#define _NBL_GLSL_BLIT_INCLUDED_
+
+#ifndef _NBL_GLSL_BLIT_MAIN_DEFINED_
+
+#include <nbl/builtin/glsl/blit/parameters.glsl>
+nbl_glsl_blit_parameters_t nbl_glsl_blit_getParameters();
+
+vec4 nbl_glsl_blit_getData(in vec3 texCoord, in uint layerIdx);
+void nbl_glsl_blit_setData(in vec4 data, in uvec3 coord, in uint layerIdx);
+vec4 nbl_glsl_blit_getKernelWeight(in uint index);
+void nbl_glsl_blit_addToHistogram(in uint bucketIndex, in uint layerIdx);
+
+#ifndef _NBL_GLSL_BLIT_ALPHA_BIN_COUNT_
+#error _NBL_GLSL_BLIT_ALPHA_BIN_COUNT_ must be defined
+#endif
+
+#define scratchShared _NBL_GLSL_SCRATCH_SHARED_DEFINED_
+
+uvec3 linearIndexTo3DIndex(in uint linearIndex, in uvec3 gridDim)
+{
+	uvec3 index3D;
+
+	const uint areaXY = gridDim.x * gridDim.y;
+	index3D.z = linearIndex / areaXY;
+	index3D.y = (linearIndex - (index3D.z * areaXY)) / gridDim.x;
+	index3D.x = linearIndex - (index3D.y*gridDim.x) - (index3D.z*areaXY);
+
+	return index3D;	
+}
+
+uint nbl_glsl_multi_dimensional_array_addressing_snakeCurve(in uvec3 coordinate, in uvec3 extents)
+{
+	return (coordinate.z * extents.y + coordinate.y) * extents.x + coordinate.x;
+}
```

Comment:
move to some utility header (`multi_dimensional_array_addressing.glsl`?)

Reply:
Resolved.